packages/cosmic-swingset/src/computron-counter.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+
+import { assert } from '@endo/errors';
+import {
+  BeansPerBlockComputeLimit,
+  BeansPerVatCreation,
+  BeansPerXsnapComputron,
+} from './sim-params.js';
+
+const { hasOwn } = Object;
+
+/**
+ * @typedef {object} BeansPerUnit
+ * @property {bigint} blockComputeLimit
+ * @property {bigint} vatCreation
+ * @property {bigint} xsnapComputron
+ */
+
+/**
+ * Return a stateful run policy that supports two phases: first allow
+ * non-cleanup work (presumably deliveries) until an overrideable computron
+ * budget is exhausted, then (iff no work was done and at least one vat cleanup
+ * budget field is positive) a cleanup phase that allows cleanup work (and
+ * presumably nothing else) until one of those fields is exhausted.
+ * https://github.com/Agoric/agoric-sdk/issues/8928#issuecomment-2053357870
+ *
+ * @param {object} params
+ * @param {BeansPerUnit} params.beansPerUnit
+ * @param {import('@agoric/swingset-vat').CleanupBudget} [params.vatCleanupBudget]
+ * @param {boolean} [ignoreBlockLimit]
+ * @returns {import('./launch-chain.js').ChainRunPolicy}
+ */
+export function computronCounter(
```

Comment:
I'm thrilled to see this pulled out of launch-chain.js, but we actually already have packages/SwingSet/src/lib/runPolicies.js. I think merging it in would be out-of-scope for this PR because it's a cosmic-swingset [*Chain*RunPolicy](https://github.com/Agoric/agoric-sdk/blob/62aabb6253187eed43cafdb807b151bd443243d4/packages/cosmic-swingset/src/launch-chain.js#L277-L282) rather than merely a SwingSet [RunPolicy](https://github.com/Agoric/agoric-sdk/blob/62aabb6253187eed43cafdb807b151bd443243d4/packages/SwingSet/src/types-external.js#L271-L278), although in the fullness of time that's probably exactly what should happen because the extra methods of the former aren't really blockchain-specific.

At any rate, focusing on the present, I'd like to see this function renamed for accuracy, e.g.:
```suggestion
export function makeExclusiveCleanupRunPolicy(
```
(or similar).

and its containing file renamed to something like chain-run-polic{y,ies}.js.

Reply:
Thanks for all the feedback, but due to how much stuff I'm juggling, I'm inclined to leave the swingset refactoring to folks like yourself that spend more time in the swingset code.